File: dart/dart-sample/sample/io/file.dart
import 'dart:io';

void main(List<String> args) async {
  // reading files
  var hosts = File(r'C:\Windows\system32\drivers\etc\hosts');
  var content = await hosts.readAsString();
  print(content);

  var lines = await hosts.readAsLines();
  print(lines.length);

  // writing files
  var file = File('output.txt');
  var sink = file.openWrite(mode: FileMode.write);
  sink.writeln('hello world!');
  sink.writeln('now is ${DateTime.now()}');
  await sink.flush();
  await sink.close();

  // deleting files
  file.delete();
}

File: kotlin/kotlin-demo/src/main/kotlin/tech/techstay/io/File.kt
package tech.techstay.io

import java.io.File

fun main() {
    // reading the whole file
    val filename = """C:\Windows\System32\drivers\etc\hosts"""
    var file = File(filename)
    println(file.readText())

    // reading files by lines
    file.readLines().take(3).forEach(::println)

    // handling lines directly
    file.forEachLine(action = ::println)

    // writing the whole file
    file = File("test.txt")
    file.writeText("some contents.\nsome more contents.\nmore and more contents.\n")

    // appending contents to files
    file.appendText("appending contents.")

    // deleting files finally
    file.delete()

}
